temporal-sdk/src/main/java/io/temporal/internal/sync/WorkflowInternal.java

```diff
@@ -89,6 +89,11 @@ public static Promise<Void> newTimer(Duration duration) {
     return getWorkflowOutboundInterceptor().newTimer(duration);
   }
 
+  public static Promise<Void> newTimer(Duration duration, TimerOptions options) {
```

Comment:
@cretz one thing I debated on was if I should include `duration` in `TimerOptions`. We didn't in Go, but I didn't couldn't find a strong reason why?

Reply:
No strong reason. We just figured in Go (and other langs?) that having required duration in both overloaded forms makes sense, and these are just "additional" options beyond duration.